### ISSUE TITLE
Update Analytics setup to display all available accounts and properties

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#9348]
## Feature Description

The Analytics accountSummaries resource of the Admin API has a maximum limit of 200 per page. For the vast majority of accounts, this is more than enough to get summaries of all accounts and properties for a user to set up the Analytics module. In some cases, such as when using an account owned by an agency or developer, it may be possible to exceed this in a way which makes it impossible to select the appropriate account for connecting the module. To unblock this case, we should update the way account summaries are requested to support querying all available accounts and their properties. 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* A user should be able to set up the Analytics module using the same dropdown interface as today, regardless of how many accounts they have access to

## Implementation Brief

- Update the `account-summaries` GA4 datapoint handler to accept a new optional `nextPageToken` value
	- Update the response parsing to return the response, as it is returned from the API, i.e. 
		```js
		{ 
			accountSummaries : Array,
			nextPageToken : string
		}
		```
- Update the fetch store for `getAccountSummaries` to merge in the response rather than overwriting it, applying the same parsing, modifications, and sorting as previously done on the server
	- Better yet – delegate this to a new action (e.g. `transformAndSortAccountSummaries`) that can be called after all pages have been requested, then process all at once
- Update the resolver for `getAccountSummaries` to request all subsequent pages (i.e. until the response no longer has a `nextPageToken`), passing the returned `nextPageToken` to the fetch store to get its page until there are none left.
	- Finally, finish by transforming and sorting all accounts and property summaries as mentioned above

### Test Coverage

* Update tests (likely just fetch mocks) for `getAccountSummaries`
* Add test coverage for new multi-page fetch behavior

## QA Brief

This will be difficult to test for real as most Google accounts do not have access to so many Analytics accounts. As a temporary workaround, the page size requested for each request to `GET:account-summaries` can be modified to return a much smaller number (like `5` or `10`) to fully exercise the same behavior without needing so many accounts to start with.

### Steps

Note: As mentioned this is bit tricky to test, this will require temporraily changing the piece of code.

1. Refer [this line](https://github.com/google/site-kit-wp/pull/9440/files#diff-057a7c0b7d3ca6ea365f58acd6913b97a0b3bd5fa2bd139b1bb492fd4e7d1725R1053), change the `pageSize` value from 200 to `2` or `3`, so that it will be practical to test.

2. Go to "Site Kit > Settings > Analytics", click "Edit", there should be a request to "account-summaries" endpoint in browser console's Network tab. If you have total 8 accounts and pageSize is set to 2, there should be total 4 requests to account-summaries endpoint. 

3. Once all the requests are processed, you shall be able to see the accounts alphabetically sorted in accounts dropdown.

## Changelog entry

* Update Analytics setup to support any number of accounts.

**Comment by eugene-manuilov** [comment]
IB ✔️

**Comment by ankitrox** [comment]
@aaemnnosttv As per our discussion, I've assigned this task to you for code review. 
Thanks

**Comment by mohitwp** [comment]
### QA Update ❌ 

Changes in the ticket caused a regression issue, and new properties are not being created in the main and development environments, resulting in an error.

![Image](https://github.com/user-attachments/assets/941c499f-f975-47fb-a03b-d723bdf6bd4e)

https://github.com/user-attachments/assets/de1fb757-4fb3-4873-8726-e7ac72049c8c

**Comment by nfmohit** [comment]
@mohitwp I've merged the follow-up PR that solves the above issue. Back to you for another pass, thanks!

**Comment by mohitwp** [comment]
### QA Update ❌ 

- Tested on main environment.
- Tested using google a/c which have 100 analytics account.
- Change page numbers to 2,3 and 4. 
- Verified that "account-summaries" requests made as per total account numbers divide by number of pages.
- Verified that once all the requests are processed, user is able to see the accounts alphabetically sorted in accounts dropdown.

@ankitrox 
The actual acceptance criteria of this ticket has been fulfilled, but I noticed another regression due to the changes."

**Issue steps** 

- Connect Analytics.
- Go to Settings and disconnect analytics.
- Connect analytics again.
- On setup screen select 'Create new property'.
- Notice error on setup screen.
- On main dashboard success notice will appear.
- Go to settings> edit analytics.
- Notice error will display and web stream will not appear as selected. Refreshing the page also will not select the web stream.

<details>

![Image](https://github.com/user-attachments/assets/e0159694-4559-437e-925f-6cfdc8e82888)

https://github.com/user-attachments/assets/d56ca278-9156-41a1-86d0-c38fbd7807c9


</details>

## PASS CASES 

**Scenario 1**

- Ga4 Accounts number - 100
- Page numbers - 2
- Requests - 50

<details>

![Image](https://github.com/user-attachments/assets/1eb30183-3713-4664-aef8-31f81bb929b3)

![Image](https://github.com/user-attachments/assets/90c94cde-a7bb-4f40-b4be-df82fe6494b3)

https://github.com/user-attachments/assets/1505d05f-4017-4de9-9d54-d4d10f7ee62d


</details>

**Scenario 2**

- Ga4 Accounts number - 100
- Page numbers - 3
- Requests - 34

<details>

![Image](https://github.com/user-attachments/assets/b81148ec-ddc2-4964-aa47-9d69baa07bc8)

![Image](https://github.com/user-attachments/assets/1c2497b2-30f7-4ccf-8fcf-cd49b6b68dc5)

</details>


**Scenario 3**

- Ga4 Accounts number - 100
- Page numbers - 4
- Requests - 25

<details>

![Image](https://github.com/user-attachments/assets/e9cd3c5c-2ba5-4a76-915f-ff20209593bc)

![Image](https://github.com/user-attachments/assets/5bd9b636-a521-4299-a8d7-4d41b3208794)

</details>

**Comment by techanvil** [comment]
Thanks @mohitwp. I've raised a PR to fix the regression you have raised: https://github.com/google/site-kit-wp/pull/9532

**Comment by nfmohit** [comment]
@mohitwp The follow-up PR has been merged and this is now back with you for another pass. Thanks!

**Comment by mohitwp** [comment]
### QA Update ✅ 

- Tested on main environment.
- Tested using google a/c which have 100 analytics account.
- Change page numbers to 2,3 and 4. 
- Verified that "account-summaries" requests made as per total account numbers divide by number of pages.
- Verified that once all the requests are processed, user is able to see the accounts alphabetically sorted in accounts dropdown.
- Verified reported issue is resolve now.

<details>

https://github.com/user-attachments/assets/d3404cb7-33ba-45be-8f90-24b345c41432

</details>

## PASS CASES 

**Scenario 1**

- Ga4 Accounts number - 100
- Page numbers - 2
- Requests - 50

<details>

![Image](https://github.com/user-attachments/assets/1eb30183-3713-4664-aef8-31f81bb929b3)

![Image](https://github.com/user-attachments/assets/90c94cde-a7bb-4f40-b4be-df82fe6494b3)

https://github.com/user-attachments/assets/1505d05f-4017-4de9-9d54-d4d10f7ee62d


</details>

**Scenario 2**

- Ga4 Accounts number - 100
- Page numbers - 3
- Requests - 34

<details>

![Image](https://github.com/user-attachments/assets/b81148ec-ddc2-4964-aa47-9d69baa07bc8)

![Image](https://github.com/user-attachments/assets/1c2497b2-30f7-4ccf-8fcf-cd49b6b68dc5)

</details>


**Scenario 3**

- Ga4 Accounts number - 100
- Page numbers - 4
- Requests - 25

<details>

![Image](https://github.com/user-attachments/assets/e9cd3c5c-2ba5-4a76-915f-ff20209593bc)

![Image](https://github.com/user-attachments/assets/5bd9b636-a521-4299-a8d7-4d41b3208794)

</details>
